s3/replication/manager/tests/system/tag_transfer_test.py

```diff
@@ -0,0 +1,279 @@
+        self.max_obj_size = self._config["max_obj_size"]
+        self.fixed_size = self._config["fixed_size"]
+        self.random_size_enabled = self._config["random_size_enabled"]
+        self.source_bucket = self._config["source_bucket"]
+        self.target_bucket = self._config["target_bucket"]
+        self.polling_wait_time = self._config["polling_wait_time"]
+        self.polling_count = self._config["polling_count"]
+
+
+# Helper methods
+
+
+def init_logger():
+    log_config_file = os.path.join(os.path.dirname(__file__),
+                                   'config', 'logger_config.yaml')
+
+    print("Using log config {}".format(log_config_file))
+    logger = setup_logger('client_tests', log_config_file)
+    if logger is None:
+        print("Failed to configure logging.\n")
+        sys.exit(-1)
+    return logger
+
+
+def create_job_with_fdmi_record(s3_config, test_config, object_info):
+    job_dict = fdmi_record_tag_template()
+
+    # Update the fields in template.
+    # XXX Removed md5 and lenght
+    job_dict["Bucket-Name"] = test_config.source_bucket
+    job_dict["Object-Name"] = object_info["object_name"]
+    job_dict["Object-URI"] = test_config.source_bucket + \
+        '\\\\' + object_info["object_name"]
+    job_dict["User-Defined"]["x-amz-meta-target-site"] = \
+        s3_config.s3_service_name
+    job_dict["User-Defined"]["x-amz-meta-target-bucket"] = \
+        test_config.target_bucket
+
+    return job_dict
+
+
+async def async_put_object_tagging(session, bucket_name, object_name, tag_name, tag_value):
+
+    request_id = str(uuid.uuid4())
+
+    obj = S3AsyncPutObjectTagging(session, request_id, bucket_name,
+                                  object_name, tag_name, tag_value)
+
+    await obj.send()
+
+    status = "success"
+    # Start transfer
+    # if object_writer.get_state() != S3RequestState.COMPLETED:
+    #    status = "failed"
+
+    return {"object_name": object_name, "status": status}
+
+
+async def setup_source(session, test_config):
+    # Create objects at source and returns list with object info
+
+    # [{"object_name": "object-name1", "size": 4096, "md5": abcd},
+    #  {"object_name": "object-name2", "size": 8192, "md5": pqrs}]
+    put_task_list = []
+    for i in range(test_config.count_of_obj):
+        # Generate object size
+        object_size = test_config.fixed_size
+        if test_config.random_size_enabled:
+            object_size = randrange(
+                test_config.min_obj_size,
+                test_config.max_obj_size)
+
+        # Generate object name
+        object_name = "test_object_" + str(i) + "_sz" + str(object_size)
+        tag_name = "user-tag-" + str(i)
+        tag_value = "tag-value-" + str(i)
+
+        # Perform the PUT operation on source and capture md5.
+        task = asyncio.ensure_future(
+            async_put_object_tagging(session, test_config.source_bucket, object_name, tag_name, tag_value))
```

Comment:
do we support only one tag?

Reply:
No we can have multiple tags as well. You are right. We can support multiple tags put, but right now its just 1 tag in this test file.